kinder/pkg/cluster/manager/actions/smoke-test.go

```diff
@@ -66,6 +66,9 @@ func SmokeTest(c *status.Cluster, wait time.Duration) error {
 		return err
 	}
 
+	// wait a few seconds for svc to be reachable
+	time.Sleep(3 * time.Second)
```

Comment:
Is it possible to replace this with a call to kubectl wait? ( kubectl wait --for=condition=ready service/nginx --timeout=30s)

Reply:
^ +1

Reply:
Thanks for the feedback. Looks like `kubectl wait` can't wait for service to be ready yet (https://github.com/kubernetes/kubernetes/issues/80828). There is an open issue for this - https://github.com/kubernetes/kubernetes/issues/83094.

Reply:
ookkk...
Considering the smoke test command is not used in CI, I think we can go with this for now,  but please add a TODO in the code to get this fixed in the future
@neolit123 opinions?

Reply:
TODO for now seems fine.

the proper way to solve this would be to add a new function here:
https://github.com/kubernetes/kubeadm/blob/83de8de3dbb3a3bb40e17d6df3638bec4bf6f28a/kinder/pkg/cluster/manager/actions/waiter.go

```
func nodePortIsReady(c *status.Cluster, port string) func(c *status.Cluster, n *status.Node) bool {
...
}
```

and also a new `waitForNodePort(...)` (<- `checkNodePort` logic goes there) that uses `waitFor()`. see `waitForPodsRunning` for example.

Reply:
I can try to do this the proper way later today.

Reply:
I have added a waiter function. PTAL when you can. :)